File: 백준/Silver/10845. 큐/큐.cc
#define BOJ_SUBMIT
#ifndef BOJ_10000_SRC_10845_QUEUE_H_
#define BOJ_10000_SRC_10845_QUEUE_H_

#include <iostream>
#include <queue>
#include <string>
#include <vector>

class Queue {
 public:
  void SetInputs() {
    std::cin >> n;
    std::cin.ignore();
  }

  void Calculate() {
    results.clear();
    std::string command;
    for (int i = 0; i < n; ++i) {
      std::cin >> command;
      if (command == "front") {
        if (integer_queue.empty()) {
          results.push_back(-1);
        } else {
          results.push_back(integer_queue.front());
        }
      } else if (command == "back") {
        if (integer_queue.empty()) {
          results.push_back(-1);
        } else {
          results.push_back(integer_queue.back());
        }
      } else if (command == "size") {
        results.push_back(integer_queue.size());
      } else if (command == "empty") {
        results.push_back(integer_queue.empty());
      } else if (command == "pop") {
        if (integer_queue.empty()) {
          results.push_back(-1);
        } else {
          results.push_back(integer_queue.front());
          integer_queue.pop();
        }
      } else if (command == "push") {
        int x;
        std::cin >> x;
        integer_queue.push(x);
      }
    }
  }

  void PrintResult() {
    for (const auto& result : results) {
      std::cout << result << '\n';
    }
  }

 private:
  std::queue<int> integer_queue;
  int n;
  std::vector<int> results;
};

#ifdef BOJ_SUBMIT
int main() {
  Queue queue;
  queue.SetInputs();
  queue.Calculate();
  queue.PrintResult();
  return 0;
}
#endif  // BOJ_SUBMIT

#endif  // BOJ_10000_SRC_10845_QUEUE_H_

File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

n = int(sys.stdin.readline().strip())
queue = deque()
results = []

for _ in range(n):
    command = sys.stdin.readline().strip().split()
    
    if command[0] == "push":
        queue.append(int(command[1]))
    elif command[0] == "pop":
        results.append(queue.popleft() if queue else -1)
    elif command[0] == "size":
        results.append(len(queue))
    elif command[0] == "empty":
        results.append(0 if queue else 1)
    elif command[0] == "front":
        results.append(queue[0] if queue else -1)
    elif command[0] == "back":
        results.append(queue[-1] if queue else -1)

sys.stdout.write("\n".join(map(str, results)) + "\n")
